File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260915-110924-39025/workdir/src/main/java/customerimporter/TransientCustomerSystem.java
package customerimporter;

import java.util.ArrayList;
import java.util.List;
import java.util.NoSuchElementException;

public class TransientCustomerSystem extends CustomerSystem {

    private List<Customer> customers;

    // system lifecycle

    public void start() {
        customers = new ArrayList<>();
    }

    public void stop() {
        customers = null;
    }

    // transactions

    public void beginTransaction() {

    }

    public void commit() {

    }

    // customers

    public void add(Customer aCustomer) {
        customers.add(aCustomer);
    }

    public Customer customerWithIdentificationType(String anIdType, String anIdNumber) {
        for (Customer aCustomer : customers) {
            if (aCustomer.isIdentifiedBy(anIdType, anIdNumber)) return aCustomer;
        }
        throw new NoSuchElementException("Object is not in the collection.");
    }

    public int numberOfCustomers() {
        return customers.size();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213435-24827/workdir/src/customerimporter/transient_customer_system.py
from customerimporter.customer_system import CustomerSystem


class TransientCustomerSystem(CustomerSystem):

    _customers = None

    # system lifecycle

    def start(self):
        self._customers = []

    def stop(self):
        self._customers = None

    # transactions

    def begin_transaction(self):
        pass

    def commit(self):
        pass

    # customers

    def add(self, a_customer):
        self._customers.append(a_customer)

    def customer_with_identification_type(self, an_id_type, an_id_number):
        for a_customer in self._customers:
            if a_customer.is_identified_by(an_id_type, an_id_number): return a_customer
        raise RuntimeError("Object is not in the collection.")

    def number_of_customers(self):
        return len(self._customers)
